.github/workflows/build.yml

```diff
@@ -0,0 +1,38 @@
+name: build
+on:
+  release:
+    types: [created]
+  push:
+    branches:
+      - "**"
+    paths:
+      - "header-gen/**"
+      - "**/build.yml"
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./header-gen
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "20.x"
+          registry-url: "https://npm.pkg.github.com"
+          cache: npm
+          cache-dependency-path: header-gen/package-lock.json
+      - run: npm ci
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
It's fine to keep this, but this package will likely never need to access private packages from the GH registry.